### PR TITLE
[Native][ACR-1644]update ml_training api descriptions

### PR DESCRIPTION
**Pull request by zhoonit**
Update ml_training api descriptions except property changes

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### Change Description ###

Describe your changes here.

For instance,
 - Fix broken links of native application docs.
 -  ...

### Bugs Fixed ###

Provide links to bugs here

For instance,

 - Issue #265

### API Changes ###

In case of ACR, put the link of ACR

For instance,

 - ACR: ACR-1120

### CODE REVIEW

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 1]
```diff
@@ -384,56 +391,92 @@ status = ml_train_model_set_optimizer(model, optimizer);
 There are two ways to create a dataset. One is from a file, and the other one is from a callback.
 In either case, you need to provide streams of tensor data and arrays of values representing the label, usually one-hot-encoded.
 
-#### Set a dataset from a file
+#### Sample, batch and epoch
+
+There are three concepts confusing concepts need clarification: sample, batch and epoch.
```
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: There are three confusing concepts need ~

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 2]
```diff
@@ -384,56 +391,92 @@ status = ml_train_model_set_optimizer(model, optimizer);
...
+
+Let's assum a given model requires three inputs and two labels.
```
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: assume

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 3]
```diff
@@ -384,56 +391,92 @@ status = ml_train_model_set_optimizer(model, optimizer);
...
+
+A sample denotes a single pair of inputs/labels.
 
-To create a `dataset` from a file, a training set and a label set must be provided. A validation set and a test set can be provided if needed.
+`[input_1][input_2][input_3][label_1][label_2]`
 
-A label set is a file that has actual labels each in a line. For example:
+2. Batch
+
+A batch is a bundle of samples which will be fed to a single iteration If a batch is of size 5, the batch layout will be:
```
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: missing period

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 4]
```diff
@@ -443,14 +486,15 @@ Creating a dataset from a generator function is also possible.
...
     ```c
     /**
-    * @brief      get data which size is batch for train
-    * @param[out] data
-    * @param[out] label
-    * @param[out] last if the data is finished
+    * @brief      fill a single sample
+    * @see ml_train_datagen_cb for details
+    * @param[in/out] array of allocated input buffers ready to be filled
+    * @param[in/out] array of allocated list buffers ready to be filled
```
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: list => label

File: docs/application/native/guides/machine-learning/machine-learning-training.md [hunk 5]
```diff
@@ -461,26 +505,23 @@ Creating a dataset from a generator function is also possible.
...
     /// assuming user_data is retrived from `get_user_data()`
     custom_type * user_data = get_user_data();
-    int status = ML_ERROR_NONE;
-    ml_train_dataset_h dataset;
 
-    // validation and test callback can be omitted.
-    ml_train_dataset_create_with_generator(&dataset, get_train_data,
-                                                    NULL, NULL);
-    status = ml_train_dataset_set_property(dataset, "user_data", (void *)user_data, NULL);
+    // a file data provider to be used for training
```
Comment (lhs8928) [inline docs/application/native/guides/machine-learning/machine-learning-training.md]: file data provider => generator

**Review by lhs8928 — APPROVED** [review]
LGTM
